src/qat/purr/compiler/instructions.py

```diff
@@ -347,11 +349,14 @@ def _check_filter(self, filter):
         if filter:
             for target in self.quantum_targets:
                 if isinstance(target, PulseChannel):
-                    dt = target.physical_channel.sample_time
-                    if not np.isclose(filter.duration, dt * (self.time // dt), atol=1e-12):
+                    if not np.isclose(
+                        filter.duration,
+                        calculate_duration(self, return_samples=False),
+                        atol=1e-9,
+                    ):
                         raise ValueError(
                             f"Filter duration '{filter.duration}' must be equal to Acquire "
-                            f"duration '{self.time}'."
+                            f"duration '{calculate_duration(self, return_samples=False)}'."
```

Comment:
put this in a variable `duration_nanos = calculate_duration(self, return_samples=False)` and reuse here